src/test/java/org/folio/auth/authtokenmodule/AuthTokenTest.java

```diff
@@ -1023,4 +1042,168 @@ private String getMagicPermission(String endpoint) {
     return String.format("%s.execute", Base64.encode(endpoint));
   }
 
+  @Test
+  public void testStoreSaveRefreshToken(TestContext context) {
+    var ts = new RefreshTokenStore(vertx, tenant);
+    Async async = context.async();
+    var rt = new RefreshToken(tenant, "jones", userUUID, "http://localhost:" + port);
+    ts.saveToken(rt).onComplete(context.asyncAssertSuccess(x -> {
+      ts.checkTokenNotRevoked(rt).onComplete(context.asyncAssertSuccess(y -> {
+        async.complete();
+      }));
+    }));
+  }
+
+  @Test
+  public void testStoreRefreshTokenNotFound(TestContext context) {
+    var ts = new RefreshTokenStore(vertx, tenant);
+    Async async = context.async();
+    // A RefreshToken which doesn't exist is treated as revoked.
+    var unsavedToken = new RefreshToken(tenant, "jones", userUUID, "http://localhost:" + port);
+    ts.checkTokenNotRevoked(unsavedToken).onComplete(context.asyncAssertFailure(y -> {
+      async.complete();
+    }));
+  }
+
+  @Test
+  public void testStoreSaveApiToken(TestContext context) {
+    var ts = new ApiTokenStore(vertx, tenant, tokenCreator);
+    Async async = context.async();
+    var apiToken = new ApiToken(tenant);
+    ts.saveToken(apiToken).onComplete(context.asyncAssertSuccess(x -> {
+      ts.checkTokenNotRevoked(apiToken).onComplete(context.asyncAssertSuccess(y -> {
+        async.complete();
+      }));
+    }));
+  }
+
+  @Test
+  public void testStoreApiTokenNotFound(TestContext context) {
+    var ts = new ApiTokenStore(vertx, tenant, tokenCreator);
+    Async async = context.async();
+    // A ApiToken which doesn't exist in storage is treated as revoked.
+    var unsavedToken = new ApiToken(tenant);
+    ts.checkTokenNotRevoked(unsavedToken).onComplete(context.asyncAssertFailure(x -> {
+      async.complete();
+    }));
+  }
+
+  @Test
+  public void testApiTokenRevoked(TestContext context) {
+    var ts = new ApiTokenStore(vertx, tenant, tokenCreator);
+    Async async = context.async();
+    var apiToken = new ApiToken(tenant);
+    ts.saveToken(apiToken).onComplete(context.asyncAssertSuccess(a -> {
+      ts.checkTokenNotRevoked(apiToken).onComplete(context.asyncAssertSuccess(b -> {
+        ts.revokeToken(apiToken).onComplete(context.asyncAssertSuccess(c -> {
+          ts.checkTokenNotRevoked(apiToken).onComplete(context.asyncAssertFailure(d -> {
+            async.complete();
+          }));
+        }));
+      }));
+    }));
```

Comment:
```
    ts.saveToken(apiToken)
    .compose(x -> ts.checkTokenNotRevoked(apiToken))
    .compose(x -> ts.revokeToken(apiToken))
    .onComplete(context.asyncAssertSuccess())
    .compose(x -> ts.checkTokenNotRevoked(apiToken))
    .onComplete(context.asyncAssertFailure());
```

Reply:
This is great. Yes much better that way. Thanks for taking a look Julian!